packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -375,6 +381,34 @@ contract OPContractsManager is ISemver {
         return address(uint160(bytes20(bytes.concat(versionByte, first19Bytes))));
     }
 
+    // @notice Helper method for computing a salt for deploying proxy contracts.
+    // Proxy contract salts are computed differently and include the name of the implementation contract.
+    // This ensures that the resultant address from CREATE2 is unique, as multiple proxy contracts are deployed
+    // in a single deployment.
+    function computeProxyContractSalt(
+        uint256 _l2ChainId,
+        string memory _saltMixer,
+        string memory _contractName
+    )
+        internal
+        pure
+        returns (bytes32)
+    {
+        return keccak256(abi.encode(_l2ChainId, _saltMixer, _contractName));
+    }
+
+    // @notice Helper method for computing a salt for deploying non-proxy contracts.
+    function computeNonProxiedContractSalt(
+        uint256 _l2ChainId,
+        string memory _saltMixer
+    )
+        internal
+        pure
+        returns (bytes32)
+    {
+        return keccak256(abi.encode(_l2ChainId, _saltMixer));
+    }
```

Comment:
On second thought, should we just remove this version and always include the contract name, i.e. rename `computeProxyContractSalt` to just `computeSalt`? It removes a path and just adds some consistency, feels simpler

Reply:
I'm in favor of doing that.

Reply:
done: https://github.com/ethereum-optimism/optimism/pull/13273/commits/a3b613558f1f5d1eaec90bfa208ed7ba6332e28c